mono/metadata/oop.c

```diff
@@ -69,10 +69,11 @@ static OutOfProcessMono g_oop = { NULL, NULL };
 
 void read_exception(const void* address, gsize size)
 {
+    g_assert(g_oop.readException);
     if (g_oop.readException)
         g_oop.readException(address, size, g_oop.userData);
     else
-        abort();
+        eg_unreachable();
```

Comment:
Generally, `g_assert_not_reached` is used instead of `eg_unreachable`. I would prefer that.